sw/host/opentitanlib/src/io/jtag.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use std::rc::Rc;
+use structopt::StructOpt;
+use thiserror::Error;
+use std::path::PathBuf;
+
+use crate::app::TransportWrapper;
+use crate::dif::lc_ctrl::{DifLcCtrlState, LcCtrlReg};
+use crate::impl_serializable_error;
+
+#[derive(Debug, StructOpt, Clone)]
+pub struct JtagParams {
+    #[structopt(long, help = "OpenOCD binary path.", default_value = "openocd")]
+    pub openocd: PathBuf,
+
+    #[structopt(long, help = "Path to OpenOCD JTAG adapter config file.")]
+    pub openocd_adapter_config: Option<String>,
```

Comment:
could we use a default file paths here? if its too complicated, we can address in a follow up too.